### PR TITLE
Backport fingerprint written by the Rumble people

### PR DESCRIPTION
**Pull request by jvoisin**
Taken from https://github.com/rapid7/recog/compare/master...RumbleDiscovery:main

Co-authored-by: hdm
Co-authored-by: lpar

### CODE REVIEW

**Comment by hdm** [comment]
Hello! @jvoisin offered to PR our backlog and we are happy for the help.

File: xml/snmp_sysdescr.xml [hunk 1]
```diff
@@ -1796,6 +1796,20 @@ Copyright (c) 1995-2005 by Cisco Systems
...
+  <!--======================================================================
+                                Cumulus
+   =======================================================================-->
+
+  <fingerprint pattern="^Cumulus Linux (\S+) \(Linux Kernel ([^\)]+)\)">
+    <description>Cumulus Linux</description>
+    <example os.version="3.7.2" os.version.version="4.1.33-1+cl3u17">Cumulus Linux 3.7.2 (Linux Kernel 4.1.33-1+cl3u17)</example>
+    <param pos="0" name="os.vendor" value="Cumulus"/>
+    <param pos="0" name="os.device" value="Switch"/>
+    <param pos="0" name="os.product" value="Linux"/>
```
Comment (tsellers-r7) [inline xml/snmp_sysdescr.xml:1808]: Changing `os.product` to `Cumulus Linux` allows a CPE to be generated. Can we change the value here or should we add a remap?

File: xml/ssh_banners.xml [hunk 2]
```diff
@@ -888,9 +888,10 @@
     <param pos="0" name="os.cpe23" value="cpe:/o:canonical:ubuntu_linux:13.04"/>
   </fingerprint>
 
-  <fingerprint pattern="^OpenSSH_(6\.2p2) (Ubuntu-6unbuntu\d(?:\.\d)?)$">
+  <fingerprint pattern="^OpenSSH_(6\.2p2) (Ubuntu-6\S*)$">
     <description>OpenSSH running on Ubuntu 13.10</description>
     <example service.version="6.2p2" openssh.comment="Ubuntu-6unbuntu0.4">OpenSSH_6.2p2 Ubuntu-6unbuntu0.4</example>
```
Comment (tsellers-r7) [inline xml/ssh_banners.xml:893]: Interesting, the original pattern and example look like they contain a typo  uNbuntu `6unbuntu`.

**Review by tsellers-r7 — CHANGES_REQUESTED** [review]
I've left some feedback.  On many of these I'd just handle this as post commit fixup but in this case I'd want feedback from @hdm since some of the changes might impact his production code.

Also, please run `ruby bin/recog_cleanup` to address whitespace issues.

File: xml/html_title.xml [hunk 3]
```diff
@@ -1597,6 +1605,14 @@
     <param pos="0" name="service.cpe23" value="cpe:/a:manageengine:opmanager:-"/>
   </fingerprint>
 
+  <fingerprint pattern="^ManageEngine Desktop Central 9$">
+    <description>ManageEngine Desktop Central 9</description>
+    <example>ManageEngine Desktop Central 9</example>
+    <param pos="0" name="service.vendor" value="ManageEngine"/>
+    <param pos="0" name="service.product" value="Desktop Central"/>
+    <param pos="0" name="service.cpe23" value="cpe:/a:manageengine:desktopcentral:-"/>
```
Comment (tsellers-r7) [inline xml/html_title.xml:1613]: This CPE is invalid. Was it autogenerated or manually created?

Comment (tsellers-r7) [reply xml/html_title.xml]: Just a note @jvoisin - `update_cpes.py` removes *all* CPE entries and generates only entries which are valid according to the NIST database. It does so using the `vendor` and `product` fields and uses `cpe-remap.yaml` to try to find valid values when those fields are set to something that doesn't map and where we cannot change them.

File: xml/ssh_banners.xml (hunk 2 above)
Comment (jvoisin) [reply xml/ssh_banners.xml:893]: I think it's a typo, and it should be fixed.
@hdm can you confirm?

Comment (hdm) [reply xml/ssh_banners.xml:893]: This was intentional, the older pattern didn't match all variants, this one does via \S+

File: xml/html_title.xml [hunk 4]
```diff
@@ -2764,4 +2780,91 @@
...
+    <description>ATT VPN Gateway</description>
+    <example>AT&amp;amp;T VPN Gateway</example>
+    <param pos="0" name="hw.vendor" value="ATT"/>
+    <param pos="0" name="hw.device" value="VPN"/>
+    <param pos="0" name="hw.product" value="VPN Gateway"/>
+  </fingerprint>
+
+  <fingerprint pattern="^AT&amp;amp;T U\d+: '([^']+)'$">
+    <description>ATT VPN Gateway w Hostname</description>
+    <example host.name="Austin-Boston">AT&amp;amp;T U115: 'Austin-Boston'</example>
```
[the remark this answers is not in the document]
Comment (hdm) [reply xml/html_title.xml:2813]: This can be any string (Austin-Boston was just an example based on a real-world device).

**Comment by tsellers-r7** [comment]
Thanks much @jvoisin  and @hdm !
Just an FYI, I expect to an update to Recog this afternoon or tomorrow.

**Pull request merged**